core/src/main/java/org/apache/accumulo/core/clientImpl/bulk/ConcurrentKeyExtentCache.java

```diff
@@ -120,12 +125,24 @@ public KeyExtent lookup(Text row) {
 
         for (Text lookupRow : lookupRows) {
           if (getFromCache(lookupRow) == null) {
-            Iterator<KeyExtent> iter = lookupExtents(lookupRow).iterator();
-            while (iter.hasNext()) {
-              KeyExtent ke2 = iter.next();
-              if (inCache(ke2))
+            while (true) {
+              try {
+                Iterator<KeyExtent> iter = lookupExtents(lookupRow).iterator();
+                while (iter.hasNext()) {
+                  KeyExtent ke2 = iter.next();
+                  if (inCache(ke2))
+                    break;
+                  updateCache(ke2);
+                }
                 break;
-              updateCache(ke2);
+              } catch (TabletDeletedException tde) {
+                // tablets were merged away in the table, start over and try again
+                log.debug(
+                    "While trying to obtain a tablet location for bulk import a tablet was "
+                        + "deleted.   This can be caused by a concurrent merge tablet "
+                        + "operation.  If so its of no concern, otherwise it could be a problem.",
```

Comment:
Suggest getting rid of some of these extra spaces after the periods, and other small punctuation/grammar:

```suggestion
                    "While trying to obtain a tablet location for bulk import, a tablet was "
                        + "deleted. If this was caused by a concurrent merge tablet "
                        + "operation, this is okay. Otherwise, it could be a problem.",
```